src/util-base64.c

```diff
@@ -165,6 +165,11 @@ Base64Ecode DecodeBase64(uint8_t *dest, uint32_t dest_size, const uint8_t *src,
             ecode = BASE64_ECODE_BUF;
             return ecode;
         }
+        /* if the destination size is not at least 3 Bytes long, it'll give a dynamic
+         * buffer overflow while decoding, so, return and let the caller take care of the
+         * remaining bytes to be decoded which should always be < 4 at this stage */
+        if (dest_size - *decoded_bytes < 3)
```

Comment:
We can still put `numDecoded_blk` bytes in it

Reply:
Indeed. But, not having 3 decoded bytes would overflow in https://github.com/inashivb/suricata/blob/adc4e7e9056c23a1145cb343deb433dc23ae0513/src/util-base64.c#L77. If we want to keep the numDecoded_blk bytes, we should handle such cases/change the decoding algorithm.
From what I understand, we always only want to decode a full block of data (i.e. 4 bytes).
Are you saying that we should decode even the partial data?

Reply:
> Are you saying that we should decode even the partial data?

Yes, we should be able to decode 2 bytes of base64 data even if it does not make a multiple of 3